### ISSUE TITLE
What is the appropriate amount of data to get good results ?

### ISSUE THREAD

**Issue by vince62s** [ottokart/punctuator#7]
Do we need 1MB, 10MB 100MB of training data ?

what is the exact purpose of the dev set ?

cheers,

**Comment by ottokart** [comment]
I've used about 2.7 GB of training data. The more, the better. Be prepared to train for weeks :)
Dev/validation set is used to evaluate the model (when the validation loss stops decreasing, the training stops) and the chosen hyperparamters (learning rate, model size etc...).

**Comment by vince62s** [comment]
is the model based only on the vocab file ? I mean what if the training data carries a lot of words that are not in the vocab file ?

**Comment by ottokart** [comment]
Words that don't occur in the vocabulary, are mapped to an unknown word tag ("<unk>"). The vocabulary should consist of N most frequent words in the training data (I've used N = 100 000). Or you can use words that occur at least some reasonable number of times in the training text (e.g. 5).

**Comment by vince62s** [comment]
another question : punctuation and restoring the case are 2 subjects that are tied.
should I leave things lowercased first, train a punctuator model with lowercased data, do the punctuation restoration, and restoring the case AFTER, or should I train a model with true casing and punctuate a cased text.
not sure if I am clear.

**Comment by ottokart** [comment]
Depends on your application. My application was restoring punctuation in automatic speech recognition output, so I had no knowledge about sentence boundaries. Thus I used truecased words (i.e. first words of sentences were lowercased unless they were names etc...).

**Comment by vince62s** [comment]
just to be clear that I understand.
you train your model with Truecased data.
your ASR output is what : (1) lowercased data ? (2) truecased data ?
if (1) do you post-process with a truecase.perl script first and then puntuator ?
if (2) does this mean your vocab / LM in your ASR is Truecased and not lowercased like most systems ?
in this (2) situation you just have to run your punctuator.

Am I understanding correcly ?

I am trying to figure out also how to configure my ASR for the best

thanks

**Comment by ottokart** [comment]
Sorry for the delay.
In our ASR the output is truecased text (2) and our LM vocabulary is also truecased. But this is just a choice we made. True casing can give additional information about the meaning of the text and if it's available in the trainging data with good enough quality, then why remove it? 
Regardless, the Puncutator should work on lowercased text just as well (although I have not tried it personally, I see no reason why it shouldn't).
One advantage of using truecased vocabulary is that only the first words of sentences need casing restoration and this is trivial after the punctuation restoration has been applied.

PS! I think I misunderstood your previous question, when writing my last reply. As I now understand, you were asking in what order punctuation and casing restoration should be applied in case the ASR uses lowercased vocabulary. I must admit that I don't have any experience with casing restoration, as we have not needed it so far. But I think that if I had lowercased vocabulary, I would restore the punctuation first (and train the punctuator on lowercased text) and then restore the casing.
